src/tactics/FStar.Tactics.CtrlRewrite.fs

```diff
@@ -218,6 +218,26 @@ and on_subterms
   : tac<(term * ctrl_flag)>
   = let recurse env tm = ctrl_fold_env g0 d controller rewriter env tm in
     let rr = recurse env in (* recurse on current env *)
+    let rec descend_binders orig accum_binders accum_flag env bs t rebuild =
```

Comment:
Part of the change in this PR is to enhance the CtrlRewrite (the underlying code for the pointwise tactic and variants like l_to_r) to actually descend in the binders of an lambda, a refinement type etc.

Note, while this is an improvement, it is still incomplete ... there are other parts of the syntax of terms that this does not descend into. 

As a separate PR, I will provide a more complete implementation of this function